applications/ChimeraApplication/python_scripts/fluid_chimera_analysis.py

```diff
@@ -0,0 +1,158 @@
+                if level_parameters.Has("internal_parts_for_chimera"):
+                    part_name_list = level_parameters["internal_parts_for_chimera"]
+                    for part_name in part_name_list:
+                        self.chimera_internal_parts.append(part_name.GetString())
+
+        self.reformulate_every_step = False
+        if self.chimera_parameters.Has("reformulate_chimera_every_step"):
+            self.reformulate_every_step = self.chimera_parameters["reformulate_chimera_every_step"].GetBool()
+
+        if not self.fluid_parameters.Has("reform_dofs_at_each_step"):
+            self.fluid_parameters.AddEmptyValue("reform_dofs_at_each_step")
+
+        self.fluid_parameters["reform_dofs_at_each_step"].SetBool(self.reformulate_every_step)
+
+        # Import parallel modules if needed
+        # has to be done before the base-class constuctor is called (in which the solver is constructed)
+        if (parameters["problem_data"]["parallel_type"].GetString() == "MPI"):
+            raise Exception("MPI-Chimera is not implemented yet")
+
+        super(FluidChimeraAnalysis,self).__init__(model,self.full_parameters)
+
+    def Initialize(self):
+        super(FluidChimeraAnalysis,self).Initialize()
+        self.__SetChimeraInternalPartsFlag()
+
+    def _CreateProcesses(self, parameter_name, initialization_order):
+
+        list_of_processes = super(FluidChimeraAnalysis,self)._CreateProcesses(parameter_name, initialization_order)
+
+        main_model_part = self.model[self.fluid_parameters["model_part_name"].GetString()]
+        domain_size = main_model_part.ProcessInfo[KratosMultiphysics.DOMAIN_SIZE]
+        solver_type = self.fluid_parameters["solver_type"].GetString()
+
+
+        # Creating the necessary variant of the apply chimera process.
+        if domain_size == 2:
+            if(solver_type == "Monolithic" or solver_type == "monolithic" or solver_type == "chimera_embedded"):
+                self.chimera_process = KratosChimera.ApplyChimeraProcessMonolithic2d(main_model_part,self.chimera_levels)
+            elif (solver_type == "fractional_step" or solver_type == "FractionalStep"):
+                self.chimera_process = KratosChimera.ApplyChimeraProcessFractionalStep2d(main_model_part,self.chimera_levels)
+        else:
+            if(solver_type == "Monolithic" or solver_type == "monolithic" or solver_type == "chimera_embedded"):
+                self.chimera_process = KratosChimera.ApplyChimeraProcessMonolithic3d(main_model_part,self.chimera_levels)
+            elif (solver_type == "fractional_step" or solver_type == "FractionalStep"):
+                self.chimera_process = KratosChimera.ApplyChimeraProcessFractionalStep3d(main_model_part,self.chimera_levels)
+
+        self.chimera_process.SetEchoLevel(self.chimera_echo_lvl)
+        self.chimera_process.SetReformulateEveryStep(self.reformulate_every_step)
+
+        return list_of_processes
+
+    def __SetChimeraInternalPartsFlag(self):
```

Comment:
I'm wondering if we should do this at the solver level...

Reply:
I'm aware that this would imply to modify your current json structure to store the Chimera stuff inside the `solver_settings`. After my experience with the embedded solver and the level set calculation and correction I think this is a more clean solution.

Reply:
I do not need to do change it ! Any way now they are done in the solvers using an external utility module !